ext/tidy/tests/parsing_file_too_large.phpt

```diff
@@ -0,0 +1,56 @@
+--TEST--
+Trying to parse a file that is too large (over 4GB)
+--EXTENSIONS--
+tidy
+--SKIPIF--
+<?php
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--INI--
+memory_limit="5G"
+--FILE--
+<?php
+
+$path = __DIR__ . '/too_large_test.html';
+$file = fopen($path, 'w+');
+
+// Write over 4GB
+const MIN_FILE_SIZE = 4_294_967_295;
+$total_bytes = 0;
+$s = str_repeat("a", 10_000);
+while ($total_bytes < MIN_FILE_SIZE) {
+    $bytes_written = fwrite($file, $s);
+    if ($bytes_written === false) {
+        echo "Didn't write bytes\n";
+    }
+    $total_bytes += $bytes_written;
+}
```

Comment:
Is a sparse file sufficient to reproduce the issue? You should be able to create one by seeking to a large offset and then writing something. This should be much faster and take up less disk storage when running the test.

Reply:
I didn't know about sparse files until now. Let me try that.

Reply:
The test LGTM now. I trust you tested that it still does what you expect it to do.

Reply:
I did :)